pkg/metrics/metrics.go

```diff
@@ -242,6 +254,56 @@ func NewServerMetrics() *ServerMetrics {
 	}
 }
 
+func NewResticServerMetrics() *ServerMetrics {
+	return &ServerMetrics{
+		metrics: map[string]prometheus.Collector{
+			podVolumeBackupEnqueueTotal: prometheus.NewCounterVec(
+				prometheus.CounterOpts{
+					Namespace: resticMetricsNamespace,
+					Name:      podVolumeBackupEnqueueTotal,
+					Help:      "Total number of pod_volume_backup objects enqueued",
+				},
+				[]string{nodeMetricLabel},
+			),
+			podVolumeBackupDequeueTotal: prometheus.NewCounterVec(
+				prometheus.CounterOpts{
+					Namespace: resticMetricsNamespace,
+					Name:      podVolumeBackupDequeueTotal,
+					Help:      "Total number of pod_volume_backup objects dequeued",
+				},
+				[]string{nodeMetricLabel},
+			),
+			resticOperationLatencyGaugeSeconds: prometheus.NewGaugeVec(
+				prometheus.GaugeOpts{
+					Namespace: resticMetricsNamespace,
+					Name:      resticOperationLatencyGaugeSeconds,
+					Help:      "Gauge metric indicating time taken, in seconds, to perform restic operations",
+				},
+				[]string{nodeMetricLabel, resticOperationLabel, backupNameLabel, pvbNameLabel},
+			),
+			resticOperationLatencySeconds: prometheus.NewHistogramVec(
+				prometheus.HistogramOpts{
+					Namespace: resticMetricsNamespace,
+					Name:      resticOperationLatencySeconds,
+					Help:      "Time taken to complete restic operations, in seconds",
+					Buckets: []float64{
+						toSeconds(1 * time.Minute),
+						toSeconds(5 * time.Minute),
+						toSeconds(10 * time.Minute),
+						toSeconds(15 * time.Minute),
+						toSeconds(30 * time.Minute),
+						toSeconds(1 * time.Hour),
+						toSeconds(2 * time.Hour),
+						toSeconds(3 * time.Hour),
+						toSeconds(4 * time.Hour),
```

Comment:
What happens if an operation takes longer than 4 hours? Does it go into the 4 hour bucket, or is a new one made?

Reply:
In this case 10 buckets are created for the following ranges:
1. 0-60s
1. 0-300s
1. 0-600s
1. 0-900s
1. 0-1800s
1. 0-3600s
1. 0-7200s
1. 0-10800s
1. 0-14400s
1. 0-+Inf

All the operations taking longer than 4h will fall into the last bucket [0-+Inf]

Reply:
@ashish-amarnath please feel free to merge the PR :)